packages/@react-stately/selection/docs/selection.mdx

```diff
@@ -0,0 +1,133 @@
+single key instead of a Set as their value, and `onSelectionChange` is also called with a single key.
+
+```tsx
+let [selectedKey, setSelectedKey] = useState(null);
+
+<ComboBox selectedKey={selectedKey} onSelectionChange={setSelectedKey}>
+  <Item uniqueKey="one">One</Item>
+  <Item uniqueKey="two">Two</Item>
+  <Item uniqueKey="three">Three</Item>
+</ComboBox>
+```
+
+In components which support multiple selection, you can limit the selection to a single item using the
+`selectionMode` prop. This continues to accept `selectedKeys` and `defaultSelectedKeys` as a Set, but it will
+only contain a single key at a time.
+
+```tsx
+let [selectedKeys, setSelectedKeys] = useState(new Set());
+
+<ListBox 
+  selectionMode="single"
+  selectedKeys={selectedKeys}
+  onSelectionChange={setSelectedKeys}>
+  <Item uniqueKey="one">One</Item>
+  <Item uniqueKey="two">Two</Item>
+  <Item uniqueKey="three">Three</Item>
+</ListBox>
+```
+
+## Select All
+
+Some components support a checkbox to select all items in the collection, or a keyboard shortcut like <kbd>⌘ Cmd</kbd> + <kbd>A</kbd>.
+This represents a selection of all items in the collection, regardless of whether or not all items have been loaded from the network.
+For example, when using a component with infinite scrolling support, the user will be unaware that all items are not yet loaded because
+it loads more transparently to them as they scroll down. For this reason, it makes sense for select all to represent all items, not just
+the loaded ones.
+
+When a select all event occurs, a special flag is set on the selection object: `isSelectAll`. When this is set, the selection represents
+all items in the collection, whether currently loaded or not. The application must adjust its handling of bulk actions in this case
+to apply to the entire collection rather than only the keys available to it locally.
+
+TODO: example
+
+## Dynamic data
+
+When data in a collection changes, the selection state may need to be updated accordingly. For example, if a selected item is
+deleted, it should be removed from the set of selected keys. You can do this yourself, but the <TypeLink links={dataDocs.links} type={dataDocs.exports.useListData} />
+and <TypeLink links={dataDocs.links} type={dataDocs.exports.useTreeData} /> hooks can handle this automatically.
+
+```tsx
+let list = useListData({
+  initialItems: [
+    {name: 'Aardvark'},
+    {name: 'Kangaroo'},
+    {name: 'Snake'}
+  ],
+  initialSelectedKeys: ['Kangaroo'],
+  getKey: item => item.name
+});
+
+function removeItem() {
+  // Removing the list item will also remove it from the selection state.
+  list.remove('Kangaroo');
+}
+
+<ListBox
+  items={list.items}
+  itemKey="name"
+  selectedKeys={list.selectedKeys}
+  onSelectionChange={list.setSelectedKeys}>
+  {item => <Item>{item.name}</Item>}
+</ListBox>
+```
+
+For more information, see [useListData](useListData.html) and [useTreeData](useTreeData.html).
```

Comment:
I guess we could also have a section on async data, and setting selection before it's loaded being supported?